app/src/controllers/user/sagas.js

```diff
@@ -158,7 +160,10 @@ function* addApiKey({ payload = {} }) {
         }),
       );
     }
-    yield put(fetchApiKeysAction());
+    const apiKeys = yield select((state) => apiKeysSelector(state));
+    const updatedApiKeys = [...apiKeys];
+    updatedApiKeys.unshift({ id, name: newName, created_at });
+    yield put(setApiKeysAction(updatedApiKeys));
```

Comment:
Please move this logic to the reducer.
Create a separate action handlers to address such actions.